### ISSUE TITLE
how to save best mdoel?

### ISSUE THREAD

**Issue by wziji** [scikit-optimize/scikit-optimize#937]
`
from sklearn import datasets
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

iris=datasets.load_iris()

import xgboost as xgb
from xgboost.sklearn import XGBClassifier

from skopt import BayesSearchCV
from skopt.space import Real, Categorical, Integer
import pickle

class FixedBayesSearchCV(BayesSearchCV):

    def __init__(self, estimator, search_spaces, optimizer_kwargs=None,
                n_iter=50, scoring=None, fit_params=None, n_jobs=1,
                n_points=1, iid=True, refit=True, cv=None, verbose=0,
                pre_dispatch='2*n_jobs', random_state=None,
                error_score='raise', return_train_score=False):

        self.fit_params = fit_params
        self.search_spaces = search_spaces
        self.n_iter = n_iter
        self.n_points = n_points
        self.random_state = random_state
        self.optimizer_kwargs = optimizer_kwargs
        self._check_search_space(self.search_spaces)
 
        super(BayesSearchCV, self).__init__(
                estimator=estimator, scoring=scoring, n_jobs=n_jobs, iid=iid,
                refit=refit, cv=cv, verbose=verbose, pre_dispatch=pre_dispatch,
                error_score=error_score, return_train_score=return_train_score)
 
    def _run_search(self, x):
        raise BaseException('Use newer skopt')

        
def xgb_auto_para_tuning_bayesian(model_xgb, X, Y):
    train_x, test_x, train_y, test_y = train_test_split(X, Y, train_size=0.80, random_state=0)
    opt = FixedBayesSearchCV(model_xgb, cv=3, n_points=2, n_jobs=4, verbose=1,
        search_spaces={
            'learning_rate': Real(0.008, 0.01),
            'max_depth': Integer(3, 10),
            'gamma':Real(0, 0.5),
            'min_child_weight':Integer(1, 8),
            'subsample':Real(0.6, 1.0),
            'colsample_bytree':Real(0.6, 1.0),
            'reg_alpha':Real(0, 0.5),
            'reg_lambda':Real(0, 0.5)
        },
         fit_params={
                 'eval_set': [(test_x, test_y)],
                 'eval_metric': 'mlogloss',
                 'early_stopping_rounds': 50
                 })
    
    opt.fit(train_x, train_y)
    
    
    print("val. score: %s" % opt.best_score_)
    print("test score: %s" % opt.score(test_x, test_y))
    print("Best parameters: ", opt.best_params_)
    print("Best estimator:", opt.best_estimator_)
    
    test_y_pred = opt.predict(test_x)
    print("\nclassification_report")
    print(classification_report(test_y, test_y_pred))
    
    return opt

if __name__ == '__main__':
    iris_X=iris.data
    iris_y=iris.target
    
    print(type(iris_X[0][0]))
    print(iris_X.shape)
    print(iris_y.shape)

    model_xgb = XGBClassifier(
                learning_rate=0.01,
                n_estimators=5,
                max_depth=6,
                min_child_weight=1, 
                gamma=0, 
                subsample=0.8,
                colsample_bytree=0.8,
                objective='multi:softmax',
                num_class=3,
                scale_pos_weight=1,
                seed=1)

    opt = xgb_auto_para_tuning_bayesian(model_xgb, iris_X, iris_y)
`
